web-scraping.ipynb

```diff
@@ -0,0 +1,308 @@
+    "\n",
+    "```html\n",
+    "...\n",
+    "<div class=\"grid__item one-sixth lap-one-whole palm-one-whole\">\n",
+    "<div class=\"filter-button-group\">\n",
+    "<a href='#all' class=\"btn\">All (444)</a>\n",
+    "<a href='#featured' class=\"btn featured\">Featured</a>\n",
+    "<a href='#alarm' class=\"btn\">\n",
+    "Alarm\n",
+    "(9)\n",
+    "</a>\n",
+    "...\n",
+    "```\n",
+    "\n",
+    "The line `<a href='#all' class=\"btn\">All (444)</a>` contains the counter. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 145,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "URL = 'https://home-assistant.io/components/'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With `requests` the website is retrieved and with `BeautifulSoup` parsed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 146,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "raw_html = requests.get(URL).text\n",
+    "raw_data = BeautifulSoup(raw_html, 'html.parser')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now you have the complete content of the page. [CSS selectors](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#css-selectors) can be used to identify the counter. We have several options to get the part in question. As `BeautifulSoup` is giving us a list with the findings, we only need to identify the position in the list."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 162,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<a class=\"btn\" href=\"#all\">All (444)</a>\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(raw_data.select('a')[7])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 161,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[<a class=\"btn\" href=\"#all\">All (444)</a>]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(raw_data.select('a:nth-of-type(8)'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 140,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[<a class=\"btn\" href=\"#all\">All (444)</a>]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(raw_data.select('a[href=\"#all\"]'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 141,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<a class=\"btn\" href=\"#all\">All (444)</a>\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(raw_data.select('.btn')[0])"
```

Comment:
You should stick with finding the text using `#all`. It is too easy for a new button to get added to the page and this will break. The odds of the href changing from `#all` are very slim. When scraping you got to try to make the selectors as robust as possible.